src/middlewared/middlewared/plugins/kubernetes_linux/deployments.py

```diff
@@ -16,12 +18,20 @@ class Config:
     @filterable
     async def query(self, filters, options):
         async with api_client() as (api, context):
-            deployments = [d.to_dict() for d in (await context['apps_api'].list_deployment_for_all_namespaces()).items]
-            events = await self.middleware.call(
-                'kubernetes.get_events_of_resource_type', 'Deployment', [d['metadata']['uid'] for d in deployments]
-            )
-            for deployment in deployments:
-                deployment['events'] = events[deployment['metadata']['uid']]
+            if len(filters) == 1 and len(filters[0]) == 3 and all(
+                a == d for a, d in zip(filters[0][:2], ['metadata.namespace', '='])
```

Comment:
`filters[0][:2] == ['metadata.namespace', '=']`

Reply:
That does not cover the case when the filter is a tuple

Reply:
I should typecast it :)